docs/core/magnetic_field.md

```diff
@@ -0,0 +1,228 @@
+# Magnetic field
+
+The magnetic field component of ACTS provides functionality to describe
+arbitrary magnetic field configurations in an experiment. It is implemented in
+a generic way and can be extended to connect to an experiment specific upstream
+source of field data.
+
+Algorithms which need magnetic field information (e.g.
+{class}`Acts::AtlasStepper`, {class}`Acts::EigenStepper`) accept the magnetic
+field as an explicit argument. 
+
+## Provider interface
+
+All magnetic field implementations inherit and implement the interface {class}`Acts::MagneticFieldProvider`:
+
+
+It provides a generic interface over different implementations. To speed up
+magnetic field lookup, each implementation can have a *cache* object, which the
```

Comment:
can explicitly say when this is useful (i.e. in checking the field in adjacent /nearby area)?

Reply:
Added a line on this.